tests/playwright-test/golden.spec.ts

```diff
@@ -74,26 +122,101 @@ test('should write missing expectations locally', async ({runInlineTest}, testIn
       });
     `
   }, {}, { CI: '' });
+
   expect(result.exitCode).toBe(1);
-  expect(result.output).toContain('snapshot.txt is missing in snapshots, writing actual');
-  const data = fs.readFileSync(testInfo.outputPath('a.spec.js-snapshots/snapshot.txt'));
+  const snapshotOutputPath = testInfo.outputPath('a.spec.js-snapshots/snapshot.txt');
+  expect(result.output).toContain(`${snapshotOutputPath} is missing in snapshots, writing actual`);
+  const data = fs.readFileSync(snapshotOutputPath);
   expect(data.toString()).toBe('Hello world');
 });
 
-test('should update expectations', async ({runInlineTest}, testInfo) => {
+test('shouldn\'t write missing expectations locally for negated matcher', async ({runInlineTest}, testInfo) => {
   const result = await runInlineTest({
-    'a.spec.js-snapshots/snapshot.txt': `Hello world`,
     'a.spec.js': `
       const { test } = pwt;
       test('is a test', ({}) => {
-        expect('Hello world updated').toMatchSnapshot('snapshot.txt');
+        expect('Hello world').not.toMatchSnapshot('snapshot.txt');
+      });
+    `
+  }, {}, { CI: '' });
+
+  expect(result.exitCode).toBe(1);
+  const snapshotOutputPath = testInfo.outputPath('a.spec.js-snapshots/snapshot.txt');
+  expect(result.output).toContain(`${snapshotOutputPath} is missing in snapshots, matchers using ".not" won\'t write them automatically.`);
+  expect(fs.existsSync(snapshotOutputPath)).toBe(false);
+});
+
+test('should update snapshot with the update-snapshots flag', async ({runInlineTest}, testInfo) => {
+  const EXPECTED_SNAPSHOT = 'Hello world';
+  const ACTUAL_SNAPSHOT = 'Hello world updated';
+  const result = await runInlineTest({
+    'a.spec.js-snapshots/snapshot.txt': EXPECTED_SNAPSHOT,
+    'a.spec.js': `
+      const { test } = pwt;
+      test('is a test', ({}) => {
+        expect('${ACTUAL_SNAPSHOT}').toMatchSnapshot('snapshot.txt');
+      });
+    `
+  }, { 'update-snapshots': true });
+
+  expect(result.exitCode).toBe(0);
+  const snapshotOutputPath = testInfo.outputPath('a.spec.js-snapshots/snapshot.txt');
+  expect(result.output).toContain(`${snapshotOutputPath} does not match, writing actual.`);
+  const data = fs.readFileSync(snapshotOutputPath);
+  expect(data.toString()).toBe(ACTUAL_SNAPSHOT);
+});
+
+test('shouldn\'t update snapshot with the update-snapshots flag for negated matcher', async ({runInlineTest}, testInfo) => {
+  const EXPECTED_SNAPSHOT = 'Hello world';
+  const ACTUAL_SNAPSHOT = 'Hello world updated';
+  const result = await runInlineTest({
+    'a.spec.js-snapshots/snapshot.txt': EXPECTED_SNAPSHOT,
+    'a.spec.js': `
+      const { test } = pwt;
+      test('is a test', ({}) => {
+        expect('${ACTUAL_SNAPSHOT}').not.toMatchSnapshot('snapshot.txt');
+      });
+    `
+  }, { 'update-snapshots': true });
+
+  expect(result.exitCode).toBe(0);
+  const snapshotOutputPath = testInfo.outputPath('a.spec.js-snapshots/snapshot.txt');
```

Comment:
I wouldn't expect these files here. Everything passed just fine!

Reply:
I don't really understand what you mean here :) 
(Sorry my Engilsh isn't perfect)

Reply:
If  you want to say that `expect(data.toString()).toBe(EXPECTED_SNAPSHOT);` is unneeded, I'd argue with you because it checks that the expected snapshot wasn't updated.

Reply:
My bad, I was reading it wrong. This test is great!